dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ReferenceConfig.java

```diff
@@ -308,6 +308,7 @@ public synchronized void init() {
         }
         map.put(REGISTER_IP_KEY, hostToRegistry);
 
+        serviceMetadata.getAttachments().forEach((k, v) -> map.put(k, String.valueOf(v)));
```

Comment:
When the value is not of type String, an exception will be thrown, and it is necessary to ensure that only metadata of type String can be passed